torch/_dynamo/utils.py

```diff
@@ -961,8 +967,33 @@ def us_to_ms(field):
         metric = metrics.get(field, None)
         return metric // 1000 if metric is not None else None
 
+    def _convert_collection_to_str(field: str) -> Optional[str]:
+        def safe_str(item: Any) -> str:
+            try:
+                return str(item)
+            except Exception:
+                return str(None)
+
+        metric = metrics.get(field, None)
+        if metric is None:
+            return None
+        del metrics[field]
+        if not isinstance(metric, set) and not isinstance(metric, list):
+            return None
+        return ",".join(safe_str(item) for item in metric)
+
+    # Convert sets to strings
+    inductor_fx_remote_cache_hit_keys_str = _convert_collection_to_str(
```

Comment:
nit: Completely stylistic, so hardly worth commenting, but why not just put the call to your helper directly below?

```
"inductor_fx_remote_cache_hit_keys": _convert_collection_to_str(
```